test/jdk/javax/sound/midi/Sequencer/MetaCallback.java

```diff
@@ -31,13 +31,16 @@
 import javax.sound.midi.Sequencer;
 import javax.sound.midi.ShortMessage;
 import javax.sound.midi.Track;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
 
 /**
  * @test
  * @bug 4347135
  * @summary MIDI MetaMessage callback inconsistent
  * @key intermittent sound
- * @run main/othervm MetaCallback
+ * @run main/othervm/timeout=300 MetaCallback
```

Comment:
"On m1 mac sometimes (in very rare instance) it takes a long time to get the sequencer so increasing timeout helps with that."

Hmm. 5 minutes is a really long time for something that should take less than 1 second (getting the sequencer).
Is that because something else (perhaps another process) is holding a needed resource ?

Are any other tests running ? Or were you running this test in a loop ? Perhaps it is something like what you say happened on Linux where the test wouldn't exit until the sequencer was closed ? Suppose some other process managed to exist without properly releasing some system resource .. and the next app that needed it has to wait until the O/S cleaned up?
Is that possible ?
Our CI test harness automatically multiplies test timeout by 2  so you've now in effect got a VERY long 10 minute timeout on this test .. way too long I think.

Reply:
> Hmm. 5 minutes is a really long time for something that should take less than 1 second (getting the sequencer). Is that because something else (perhaps another process) is holding a needed resource ?

I tried to find reasons or any kind of patterns by simulating different scenarios like external applications using/holding the audio line, running other sound or midi tests in parallel, switching audio devices during the test execution - and i haven't found a way of stable reproduction or any kind of regularity in test behavior. The only thing i found out is that once test passes the normal way it never fails until system is restarted or wakes up from sleep. The only thing i can say that i had cases when a singular run of the test after the restart demonstrated this problem so it is not a resource locked by another Java instance - there are no other java processes in the system at this point. Which makes me believe that the problem is not on our side and i do not think we can do anything to circumvent the issue on our side. I will continue investigation on the OS side though.

> Our CI test harness automatically multiplies test timeout by 2 so you've now in effect got a VERY long 10 minute timeout on this test .. way too long I think.

Yikes! Yes, 5 minutes is already long enough for this test, multiplying it by two makes it unreasonably long in case when it is properly hangs. I will replace the limit to 120 which makes two minutes timeout on non-multiplied system and somewhat reasonable 4 minutes on CI.